### DISCUSSION TITLE
10.9.1 Migration deployment to new machines with DSC

### DISCUSSION THREAD

**Discussion by wmedernach** [Esri/arcgis-powershell-dsc#579] [Q&A] [answered]
We currently have a legacy 2 machine deployment (manual installs and upgrades) and I would like to migrate this to 2 new machines via DSC and I have some general questions about the process and how it relates to what we currently have configured.

Our current deployment consists of:
VM1 - Federated Hosting Server Role (Portal, Server, Data Store, 3 web adaptors (portal, arcgis, image))
VM2 - Raster Analysis Server

The idea is to setup a new 2 machine 10.9.1 environment with DSC and restore the system via WebGISDR.  Once we've confirmed the new system is functioning as expected, perform the 11.x upgrade via DSC.

Based on what I've been reading, I'm not sure this configuration can be installed with a single config json file?

I do realize that this current setup is probably not a recommended configuration, but we're not a large organization and this system has running well for 10ish years.  I'm simply looking to upgrade this via DSC to facilitate easier upgrades in the future vs manual component installs/upgrades.

Any comments/suggestions/feedback would be greatly appreciated.

**Comment by cameronkroeker** [comment, marked as answer]
Hi @wmedernach,

This setup should be achievable with 2 json configuration files. One json configuration file for the Federated Hosting Server Role (Portal, Server, Data Store, 2 web adaptors (portal, arcgis), and a second json configuration file for the Raster Analysis Server with image wa (installed on VM1 but registered with server on VM2). 

**For example:**

VM 1: https://github.com/Esri/arcgis-powershell-dsc/blob/main/SampleConfigs/v4/v4.4.0/Base%20Deployment/BaseDeployment-SingleMachine.json

VM 2: https://github.com/Esri/arcgis-powershell-dsc/blob/main/SampleConfigs/v4/v4.4.0/Gis%20Servers/GISServer-RasterAnalytics-fileshare.json
- Requires a few modifications. For example, add an additional Node for the image wa with the VM1 hostname, and ConfigData.WebAdaptor json block. This will install the image wa on VM1 but register it with the Image Server on VM2.

```json
"AllNodes": [
   {
      "NodeName": "VM2-hostname",
      "Role": [
         "Server",
         "RasterDataStoreItem"
      ]
   },
   {
      "NodeName": "VM1-hostname",
      "Role": [
         "WebAdaptor"
      ],
     "WebAdaptorConfig": [
        {
           "Role": "Server",
           "Context": "image",
           "HostName": "vm1-hostname.domain.com"
        }
      ]
   }
],
"ConfigData": {
   "Version": "10.9.1",
   "Server": {
    },
   "WebAdaptor": {
      "AdminAccessEnabled": false,
      "OverrideHTTPSBinding": false,
      "WebSiteId": 1,
      "Installer": {
         "Path": "C:\\Software\\Archives\\10.9.1\\ArcGIS_Web_Adaptor_for_Microsoft_IIS_1091_180055.exe",
         "IsSelfExtracting": true
      }
   }
}
```

Thanks,
Cameron K.

**Reply by wmedernach** [reply]
Awesome, thanks for the clarifications and samples @cameronkroeker!
